homeassistant/components/sonos/speaker.py

```diff
@@ -826,9 +826,6 @@ def async_update_groups(self, event: SonosEvent) -> None:
                     f"{SONOS_VANISHED}-{uid}",
                     reason,
                 )
-
-        if "zone_player_uui_ds_in_group" not in event.variables:
-            return
```

Comment:
Not all zone group events contain this variable, hence some of the zone group updates were not getting processed which was one cause of the system getting out of sync.